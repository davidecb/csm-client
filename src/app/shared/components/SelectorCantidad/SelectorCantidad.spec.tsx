import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { SelectorCantidad } from './';

describe('SelectorCantidad test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof SelectorCantidad> & {
    setCantidad: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      cantidad: 1,
      setCantidad: stub(),
    };
    componentWrapper = render(<SelectorCantidad {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should not allow to select quantity less than one', async () => {
    const elem = componentWrapper.container;
    const lessButton = elem.querySelectorAll('div')[1];
    
    await wait(() => {
      lessButton && fireEvent.click(lessButton);
      lessButton && fireEvent.click(lessButton);
      lessButton && fireEvent.click(lessButton);
      lessButton && fireEvent.click(lessButton);
      lessButton && fireEvent.click(lessButton);
    });
    const divCantidad = elem.querySelectorAll('div');

    expect(divCantidad.length).toBe(4);
    expect(divCantidad[2].textContent).toBe('1');
  });

  it('should show 3 before click 2 times the more button', async () => {
    const elem = componentWrapper.container;
    const moreButton = elem.querySelectorAll('div')[3];
    
    await wait(() => {
      moreButton && fireEvent.click(moreButton);
      moreButton && fireEvent.click(moreButton);
    });
    const divCantidad = elem.querySelectorAll('div');

    expect(divCantidad.length).toBe(4);
    expect(divCantidad[2].textContent).toBe('3');
  });

  it('should show 4 before click 4 times the more button and 1 time the less button', async () => {
    const elem = componentWrapper.container;
    const lessButton = elem.querySelectorAll('div')[1];
    const moreButton = elem.querySelectorAll('div')[3];
    
    await wait(() => {
      moreButton && fireEvent.click(moreButton);
      moreButton && fireEvent.click(moreButton);
      moreButton && fireEvent.click(moreButton);
      moreButton && fireEvent.click(moreButton);
      lessButton && fireEvent.click(lessButton);
    });
    const divCantidad = elem.querySelectorAll('div');

    expect(divCantidad.length).toBe(4);
    expect(divCantidad[2].textContent).toBe('4');
  });
});
