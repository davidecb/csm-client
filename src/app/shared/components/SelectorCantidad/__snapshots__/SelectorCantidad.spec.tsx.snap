// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorCantidad test should match snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm gMyFWH"
  >
    <div
      class="sc-AxirZ bgRPfF"
    >
      ➖
    </div>
    <div
      class="sc-AxiKw gKfOXm"
    >
      1
    </div>
    <div
      class="sc-AxirZ bgRPfF"
    >
      ➕
    </div>
  </div>
</div>
`;
